src/common/base/test/ICordBenchmark.cpp

```diff
@@ -0,0 +1,150 @@
+BENCHMARK_RELATIVE(cord_10k_string, iters) {
+    for (auto i = 0u; i < iters; i++) {
+        Cord cord;
+        for (int j = 0; j < 1000; j++) {
+            cord << "abcdefghij";
+        }
+        std::string str = cord.str();
+        folly::doNotOptimizeAway(&str);
+    }
+}
+
+BENCHMARK_DRAW_LINE();
+
+BENCHMARK(sstream_1k_mix, iters) {
+    for (auto i = 0u; i < iters; i++) {
+        std::stringstream ss;
+        for (int j = 0; j < 50; j++) {
+            ss << "abcdefg" << 1234567890L << true << 1.23456789;
+        }
+        std::string str = ss.str();
+        folly::doNotOptimizeAway(&str);
+    }
+}
+BENCHMARK_RELATIVE(icord_1k_mix, iters) {
+    for (auto i = 0u; i < iters; i++) {
+        ICord<> cord;
+        for (int j = 0; j < 50; j++) {
+            cord << "abcdefg"
+                 << folly::to<std::string>(1234567890L)
+                 << folly::to<std::string>(true)
+                 << folly::to<std::string>(1.23456789);
+        }
+        std::string str = cord.str();
+        folly::doNotOptimizeAway(&str);
+    }
+}
+BENCHMARK_RELATIVE(cord_1k_mix, iters) {
+    for (auto i = 0u; i < iters; i++) {
+        Cord cord;
+        for (int j = 0; j < 50; j++) {
+            cord << "abcdefg"
+                 << folly::to<std::string>(1234567890L)
+                 << folly::to<std::string>(true)
+                 << folly::to<std::string>(1.23456789);
+        }
+        std::string str = cord.str();
+        folly::doNotOptimizeAway(&str);
+    }
+}
+
+BENCHMARK_DRAW_LINE();
+
+BENCHMARK(sstream_512_mix, iters) {
+    for (auto i = 0u; i < iters; i++) {
+        std::stringstream ss;
+        for (int j = 0; j < 25; j++) {
+            ss << "abcdefg" << 1234567890L << true << 1.23456789;
+        }
+        std::string str = ss.str();
+        folly::doNotOptimizeAway(&str);
+    }
+}
+BENCHMARK_RELATIVE(icord_512_mix, iters) {
+    for (auto i = 0u; i < iters; i++) {
+        ICord<> cord;
+        for (int j = 0; j < 25; j++) {
+            cord << "abcdefg"
+                 << folly::to<std::string>(1234567890L)
+                 << folly::to<std::string>(true)
+                 << folly::to<std::string>(1.23456789);
+        }
+        std::string str = cord.str();
+        folly::doNotOptimizeAway(&str);
+    }
+}
+BENCHMARK_RELATIVE(cord_512_mix, iters) {
+    for (auto i = 0u; i < iters; i++) {
+        Cord cord;
+        for (int j = 0; j < 25; j++) {
+            cord << "abcdefg"
+                 << folly::to<std::string>(1234567890L)
+                 << folly::to<std::string>(true)
+                 << folly::to<std::string>(1.23456789);
+        }
+        std::string str = cord.str();
+        folly::doNotOptimizeAway(&str);
+    }
+}
+
+int main(int argc, char** argv) {
+    folly::init(&argc, &argv, true);
+
+    folly::runBenchmarks();
+    return 0;
+}
+
+
+/*
+// In Intel(R) Xeon(R) Platinum 8260M CPU @ 2.30GHz
+============================================================================
```

Comment:
Could you update the benchmark numbers ?

Reply:
Fixed.